src/commands/mod.rs

```diff
@@ -580,13 +580,25 @@ impl DaemonControl {
         }
     }
 
-    pub fn list_spend(&self) -> ListSpendResult {
+    pub fn list_spend(&self, txids: Option<Vec<bitcoin::Txid>>) -> ListSpendResult {
         let mut db_conn = self.db.connection();
-        let spend_txs = db_conn
-            .list_spend()
+        let spend_psbts = db_conn.list_spend();
+
+        let txids_set: Option<HashSet<_>> =
+            txids.as_ref().map(|list| list.iter().cloned().collect());
+
+        let spend_txs = spend_psbts
             .into_iter()
+            .filter(|(psbt, _)| {
+                if let Some(set) = &txids_set {
+                    set.contains(&psbt.unsigned_tx.txid())
+                } else {
+                    true
+                }
+            })
```

Comment:
nit: a `filter` followed by a `map` can be done more succintly in a single `filter_map`:
```diff
diff --git a/src/commands/mod.rs b/src/commands/mod.rs
index b54ec80..083b3a4 100644
--- a/src/commands/mod.rs
+++ b/src/commands/mod.rs
@@ -589,14 +589,14 @@ impl DaemonControl {
 
         let spend_txs = spend_psbts
             .into_iter()
-            .filter(|(psbt, _)| {
+            .filter_map(|(psbt, updated_at)| {
                 if let Some(set) = &txids_set {
-                    set.contains(&psbt.unsigned_tx.txid())
-                } else {
-                    true
+                    if !set.contains(&psbt.unsigned_tx.txid()) {
+                        return None;
+                    }
                 }
+                Some(ListSpendEntry { psbt, updated_at })
             })
-            .map(|(psbt, updated_at)| ListSpendEntry { psbt, updated_at })
             .collect();
 
         ListSpendResult { spend_txs }
```

Reply:
done